.github/workflows/reusable_docker_pipeline.yml

```diff
@@ -67,103 +108,139 @@ jobs:
+          retention-days: 1
+
+  merge_dockerhub:
+    runs-on: ubuntu-latest
+    if: inputs.publish
+    needs:
+      - docker_build
+      - prepare-metadata
     steps:
-    - name: Determine image name
-      id: set_image_name
-      run: |
-        if [ -n "${{ inputs.repoName }}" ]; then
-          echo "IMAGE_NAME=${{ inputs.repoName }}" >> $GITHUB_ENV
-        else
-          echo "IMAGE_NAME=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 2)" >> $GITHUB_ENV
-        fi
-
-    - name: Download Docker image from workspace
-      uses: actions/download-artifact@v4
-      with:
-        name: ${{ env.IMAGE_NAME }}
-        path: /tmp/
-
-    - name: Load Docker image
-      run: docker load -i /tmp/${{ env.IMAGE_NAME }}.tar
-
-    - name: Login to Docker Hub
-      uses: docker/login-action@v3
-      with:
-        username: ${{ secrets.DOCKERHUB_USERNAME }}
-        password: ${{ secrets.DOCKERHUB_TOKEN }}
-
-    - name: Push Docker image with SHA
-      run: |
-        docker tag ${{ env.IMAGE_NAME }}:${{ github.sha }} ${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ env.IMAGE_NAME }}:${{ github.sha }}
-        docker push ${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ env.IMAGE_NAME }}:${{ github.sha }}
-
-    - name: Push Docker image with Tag
-      if: startsWith(github.ref, 'refs/tags/')
-      run: |
-        docker tag ${{ env.IMAGE_NAME }}:${{ github.sha }} ${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ env.IMAGE_NAME }}:${{ github.ref_name }}
-        docker push ${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ env.IMAGE_NAME }}:${{ github.ref_name }}
-
-  ecr_publish:
-    runs-on: ubuntu-22.04
-    if: inputs.publish == true
-    needs: ["docker_build"]
+      - name: Download digests
+        uses: actions/download-artifact@v4
+        with:
+          path: /tmp/digests
+          pattern: digests-dockerhub-*
+          merge-multiple: true
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}
+          tags: |
+            type=sha,enable=true,priority=100,prefix=,suffix=,format=long
+            type=ref,enable=true,priority=200,prefix=,suffix=,event=tag
+      
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      
+      - name: Create manifest list and push
+        working-directory: /tmp/digests
+        run: |
+          tags=$(jq -cr '.tags | map("-t " + .) | join(" ")'  <<< "$DOCKER_METADATA_OUTPUT_JSON")
+          digests=$(printf "${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}@sha256:%s " $(ls /tmp/digests))
+          docker buildx imagetools create $tags $digests
+      
+      - name: Inspect image
```

Comment:
Is this step used to verify that push operation succeeded?

Reply:
Yes, this step only shows the image's information